src/components/PeopleFilters.tsx

```diff
@@ -1,12 +1,133 @@
+import { useSearchParams, Link, NavLink } from 'react-router-dom';
+
+enum Sex {
+  all = 'all',
+  mail = 'm',
+  female = 'f',
+}
+
+enum Params {
+  sex = 'sex',
+  query = 'query',
+  centuries = 'centuries',
+}
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const sex = searchParams.get(Params.sex) || '';
+  const query = searchParams.get(Params.query) || '';
+  const centuries = searchParams.getAll(Params.centuries) || '';
+
+  const handleChangeQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const params = new URLSearchParams(searchParams);
+
+    if (!event.target.value) {
+      params.delete(Params.query);
+      setSearchParams(params);
+
+      return;
+    }
+
+    params.set(Params.query, event.target.value);
+    setSearchParams(params);
+  };
+
+  const handleClickSex = (
+    event: React.MouseEvent<HTMLAnchorElement>,
+    currentSex: string,
+  ) => {
+    event.preventDefault();
+    const params = new URLSearchParams(searchParams);
+
+    if (currentSex === Sex.all) {
+      params.delete(Params.sex);
+      setSearchParams(params);
+
+      return;
+    }
+
+    params.set(Params.sex, currentSex);
+    setSearchParams(params);
+  };
+
+  const handleClickCenturies = (
+    event: React.MouseEvent<HTMLAnchorElement>,
+    currentCentury: string,
+  ) => {
+    event.preventDefault();
+
+    const params = new URLSearchParams(searchParams);
+
+    if (centuries.includes(currentCentury)) {
+      params.delete(Params.centuries);
+      centuries.forEach(century => {
+        if (century !== currentCentury) {
+          params.append(Params.centuries, century);
+        }
+      });
+
+      setSearchParams(params);
+
+      return;
+    }
+
+    params.append(Params.centuries, currentCentury);
+    setSearchParams(params);
+  };
+
+  const handleClickWithoutCenturies = (
+    event: React.MouseEvent<HTMLAnchorElement>,
+  ) => {
+    event.preventDefault();
+    searchParams.delete(Params.centuries);
+    setSearchParams(searchParams);
+  };
+
+  const handleclickDeleteFilters = (
+    event: React.MouseEvent<HTMLAnchorElement>,
+  ) => {
+    event.preventDefault();
+
+    const params = new URLSearchParams();
+
+    searchParams.forEach((value, param) => {
+      if (param === 'sort' || param === 'order') {
+        params.set(param, value);
+      }
+    });
+
+    setSearchParams(params);
+  };
+
   return (
     <nav className="panel">
       <p className="panel-heading">Filters</p>
 
       <p className="panel-tabs" data-cy="SexFilter">
-        <a className="is-active" href="#/people">All</a>
-        <a className="" href="#/people?sex=m">Male</a>
-        <a className="" href="#/people?sex=f">Female</a>
+        <Link
+          className={!sex ? 'is-active' : ''}
+          to={`/people?${searchParams.toString()}`}
+          onClick={(e) => handleClickSex(e, Sex.all)}
```

Comment:
see comment above. same for other links

```suggestion
          onClick={handleClickSex(Sex.all)}
```